nmrfx-processor-gui/src/main/java/org/nmrfx/processor/gui/log/LogDetailsView.java

```diff
@@ -0,0 +1,92 @@
+import javafx.scene.control.TextField;
+import javafx.scene.layout.ColumnConstraints;
+import javafx.scene.layout.GridPane;
+import javafx.scene.layout.Priority;
+import javafx.scene.text.Text;
+import org.nmrfx.chemistry.utilities.NvUtil;
+
+import java.awt.*;
+import java.awt.datatransfer.StringSelection;
+
+
+/**
+ * Class to display the details of a LogRecord.
+ */
+public class LogDetailsView extends GridPane {
+
+    private static final int SPACING = 5;
+    private final TextField datetime = new TextField();
+    private final TextField location = new TextField();
+    private final TextArea message = new TextArea();
+    private final Button copyButton = GlyphsDude.createIconButton(FontAwesomeIcon.COPY);
+    private String level;
+
+    public LogDetailsView() {
+        super();
+        // Set controls options
+        datetime.setEditable(false);
+        location.setEditable(false);
+        message.setEditable(false);
+        copyButton.setOnAction(this::toClipboardClicked);
+        // Add controls to GridPane (4 rows x 2 columns)
+        add(new Text("Log details: "), 0, 0, 1, 1);
+        add(copyButton, 1, 0, 1, 1);
+        add(new Text("When: "), 0, 1, 1, 1);
+        add(datetime, 1, 1, 1, 1);
+        add(new Text("Where: "), 0, 2, 1, 1);
+        add(location, 1, 2, 1, 1);
+        add(message, 1, 3, 1, 3);
+        // Set Formatting
+        setVgap(SPACING);
+        setHgap(SPACING);
+        setPadding(new Insets(SPACING, SPACING, SPACING, SPACING));
+        setHalignment(copyButton, HPos.RIGHT);
+        ColumnConstraints noConstraints = new ColumnConstraints();
+        ColumnConstraints stretchColumn = new ColumnConstraints();
+        // set Column to stretch to fill remaining horizontal space
+        stretchColumn.setHgrow(Priority.ALWAYS);
+        getColumnConstraints().addAll(noConstraints, stretchColumn);
+
+        setDetails(null);
+    }
+
+    /**
+     * Sets the controls with the values from logRecord.
+     * @param logRecord The LogRecord object.
+     */
+    public void setDetails(LogRecord logRecord) {
+        if (logRecord == null) {
+            level  = null;
+            datetime.setText("");
+            location.setText("");
+            message.setText("");
+        } else {
+            level = logRecord.getLevel().toString();
+            datetime.setText(logRecord.getTime().toString());
+            location.setText(logRecord.getLoggerName() + "#" + logRecord.getSourceMethodName());
+            message.setText(logRecord.getMessage());
+        }
+    }
+
+    /**
+     * Copies the values of the controls as well as the software version to the
+     * system clipboard.
+     * @param actionEvent
+     */
+    private void toClipboardClicked(ActionEvent actionEvent) {
+        String content = "[" + level + ":" + datetime.getText() + "] " + location.getText() + "\n" + message.getText()
+                + "\n\nSoftware version: " + NvUtil.getVersion();
```

Comment:
This returns null when running from IDE. Not a big deal, but could maybe handle the case and write "development" or something.

Reply:
Added "development" in NVUtils, not sure if thats okay or if I should only handle the case in LogDetailsView, now when running from IDE, then the window title will also say development instead of NULL
![image](https://user-images.githubusercontent.com/64663455/172425418-58f14c71-cfac-4c24-a04e-f535ddffb299.png)

also fixed NPE here, by copying empty string to clipboard if no log selected